questionary/prompts/date.py

```diff
@@ -0,0 +1,674 @@
+        """__init__ of :class: `DateCompleter`.
+
+        Offers completions for a date according to the chosen format.
+
+        Args:
+            date_format (str): Format determining the format that is to be used
+                for parsing :class: `datetime.date`.
+
+        Raises:
+            ValueError: if ``date_format`` is not in ``SUPPORTED_FORMATS``.
+        """
+        self.format: str = date_format or ISO8601
+        if self.format not in SUPPORTED_FORMATS:
+            raise (
+                ValueError(
+                    f"Date format '{self.format}' is not supported.\nSupported"
+                    f" formats:\n{SUPPORTED_FORMATS}"
+                )
+            )
+        self.delimeter: str = self.format[-3]
+
+    def _get_parse_order(self) -> List[List[str]]:
+        """Returns the order for completions.
+
+        Parses ``self.date_format`` into a list representing the order for completions,
+        e.g. [YEAR, MONTH, DAY].
+        """
+        parse_order = self.format.split(self.delimeter)
+        return [PARSE_FORMAT_DICT.get(item) for item in parse_order]  # type: ignore[misc]
+
+    def get_completions(
+        self, document: Document, complete_event: CompleteEvent
+    ) -> Iterable[Completion]:
+        """Completion of text input following a given ``date_format``.
+
+        According what has allready been typed and the given ``date_format``
+        completions for year, month and day are given (numbers only).
+        Date format ``date_format`` needs to be one of the ``SUPPORTED_FORMATS´´.
+        Supports only dates at the moment.
+
+        Args:
+            document (Document): The :class: `prompt_toolkit.document.Document` created
+                from the user input.
+            complete_event (CompleteEvent): The complete event.
+
+        Yields:
+            Iterator[Iterable[Completion]]: The completions
+        """
+        text_split = document.text.split(self.delimeter)
+        # the date information the user is currently typing, i.e. year, month or day
+        user_input = text_split[-1]
+        # old_input is the text the user has already typed
+        old_input = (
+            self.delimeter.join(text_split[:-1]) + self.delimeter
+            if len(text_split) > 1
+            else ""
+        )
+        # get correct list of completion options
+        completion_list = self._get_parse_order()[len(text_split) - 1]
+        # find fitting completions in completion_list
+        for entry in completion_list:
+            if entry.startswith(user_input):
+                completion = entry if int(entry) >= 10 else "0" + entry
+                completion += self.delimeter if len(text_split) < 3 else ""
+                yield Completion(
+                    old_input + completion,
+                    start_position=-len(document.text),
+                    style="fg:ansigreen bold",
+                    selected_style="fg:white bg:ansired bold",
```

Comment:
Can these styles be made to use one of the classes (or add a new class) so that they can be customised.

Reply:
```suggestion
style="class:options",
selected_style="class:selected",
```
where
```python
("options", "fg: ansigreen bold"),  # options given by autocomplete
```
Moreover, I introduced a default style for `selected` in `date`, which differs from the one in `DEFAULT_STYLE`:

```python
# define a default style for selected completions. Gets overwritten if another
# format for selected is given by `style`
default_date_styles = Style([("selected", "fg: white bg: ansired bold")])

merged_style = merge_styles([DEFAULT_STYLE, default_date_styles, style])
```